src/rpg_window/Window_Selectable.d.ts

```diff
@@ -0,0 +1,105 @@
+import { Window_Scrollable } from "./Window_Scrollable";
+import { Window_Help } from "./Window_Help";
+
+/**
+ * Window_Selectable
+ * 
+ * The window class with cursor movement functions.
+ */
+declare class Window_Selectable extends Window_Scrollable{
+    constructor(rect: Rectangle);
+
+    public index(): number;
+    public cursorFixed(): boolean;
+    public setCursorFixed(cursorFixed: boolean): void;
+    public cursorAll(): boolean;
+    public setCursorAll(cursorAll: boolean): void;
+    public maxCols(): number;
+    public maxItems(): number;
+    public colSpacing(): number;
+    public rowSpacing(): number;
+    public itemWidth(): number;
+    public itemHeight(): number;
+    public contentsHeight(): number;
+    public maxRows(): number;
+    public overallHeight(): number;
+    public activate(): void;
+    public deactivate(): void;
+    public select(index: number): void;
+    public forceSelect(index: number): void;
+    public smoothSelect(index: number): void;
+    public deselect(): void;
+    public reselect(): void;
+    public row(): number;
+    public topRow(): number;
+    public maxTopRow(): number;
+    public setTopRow(row: number): void;
+    public maxPageRows(): number;
+    public maxPageItems(): number;
+    public maxVisibleItems(): number;
+    public isHorizontal(): boolean;
+    public topIndex(): number;
+    public itemRect(index: number): Rectangle;
+    public itemRectWithPadding(index: number): Rectangle;
+    public itemLineRect(index: number): Rectangle;
+    public setHelpWindow(helpWindow: Window_Help): void;
+    public showHelpWindow(): void;
+    public hideHelpWindow(): void;
+    public setHandler(symbol: string, method: Function): void;
+    public isHandled(symbol: string): boolean;
+    public callHandler(symbol: string): void;
+    public isOpenAndActive(): boolean;
+    public isCursorMovable(): boolean;
+    public cursorDown(wrap: boolean): void;
+    public cursorUp(wrap: boolean): void;
+    public cursorRight(wrap: boolean): void;
+    public cursorLeft(wrap: boolean): void;
+    public cursorPagedown(): void;
+    public cursorPageup(): void;
+    public isScrollEnabled(): boolean;
+    public update(): void;
+    public processCursorMove(): void;
+
+    // TODO This method returns the values returned by processOk, processCancel, processPagedown or processPageup.
+    // All of those methods return nothing. Perhaps they're accounting for sub-classes that might override??
+    public processHandling(): any;
```

Comment:
It might be an overridable or an call back so in this matter returning unknown can be fine since it's impossible to know the return type. 
(from the Typescript Lang handbook) 
Unknown keyword : 
We may need to describe the type of variables that we do not know when we are writing an application. These values may come from dynamic content – e.g. from the user – or we may want to intentionally accept all values in our API. In these cases, we want to provide a type that tells the compiler and future readers that this variable could be anything, so we give it the unknown type.